flink-core/src/main/java/org/apache/flink/api/connector/source/SourceReader.java

```diff
@@ -132,4 +139,28 @@ default void handleSourceEvents(SourceEvent sourceEvent) {}
      */
     @Override
     default void notifyCheckpointComplete(long checkpointId) throws Exception {}
+
+    /**
+     * Pauses or resumes reading of individual source splits.
+     *
+     * <p>Note that no other methods can be called in parallel, so updating subscriptions can be
+     * done atomically. This method is simply providing connectors with more expressive APIs the
+     * opportunity to update all subscriptions at once.
+     *
+     * <p>This is currently used to align the watermarks of splits, if watermark alignment is used
+     * and the source reads from more than one split.
+     *
+     * <p>The default implementation throws an {@link UnsupportedOperationException} where the
+     * default implementation will be removed in future releases. To be compatible with future
+     * releases, it is recommended to implement this method and override the default implementation.
+     *
+     * @param splitsToPause the splits to pause
+     * @param splitsToResume the splits to resume
+     */
+    @PublicEvolving
+    default void pauseOrResumeSplits(
+            Collection<String> splitsToPause, Collection<String> splitsToResume) {
+        throw new UnsupportedOperationException(
+                "This source reader does not support pause or resume splits.");
```

Comment:
Shouldn't this be kept in sync with the other similar `throw new UnsupportedOperationException` in `SplitReader`?